apps/question/views.py

```diff
@@ -90,3 +95,24 @@ def get(self, request, **kwargs):
         'answer_ids': answer_ids,
         'question': QuestionSerializer(question).data
     })
+
+
+class QuestionFlag(APIAttemptBase):
+  """Question flagging view."""
+
+  def get(self, request, **kwargs):
+    """Get a list of all flagged question ids."""
+
+    return Response(ExamAttemptSerializer(self.attempt).data)
+
+  def post(self, request, **kwargs):
+    """Add or remove question id from flagged_questions."""
+
+    question = get_object_or_404(Question, id=self.kwargs['question_id'])
+
+    if question not in self.attempt.flagged_questions.all():
```

Comment:
You could improve this by offloading the check to the DB and doing positive check first: if self.attempt.flagged_questions.filter(id=question.id).exists():